src/common/components/MultiSelect.jsx

```diff
@@ -22,6 +25,15 @@ export default function MultipleSelectCheckmarks({
   label,
 }) {
   const classes = useStyles();
+  const [searchText, setSearchText] = useState("");
+
+  const filteredOptions = useMemo(
+    () =>
+      options.filter((option) =>
+        compareTextValue(option.label, searchText, typeof option.label)
+      ),
+    [searchText, options]
+  );
 
   useEffect(() => {
```

Comment:
Its bad way handled here, object datatype as dependency. also option.length should be dependency. please fix this code

Reply:
In this case, I want the function to re-run whenever the `options` object changes. For this use-case, adding `options.length` along might not be the best way right? What if we have 2 different arrays with the same number of items; would this useEffect run in such a scenario?

Reply:
`options` object changes on each render and useEffect is going to run each times but since there is a condition which is not going to meet everytime thus it restricts it. for now let us keep it as it is bcz see the same issue here also the with object. anyway from the working prospective it is not going to impact though.